File: solutions/c/two-fer/two_fer.c
#include <stdio.h>
#include <string.h>

#include "two_fer.h"

void two_fer(char *buffer, const char *name)
{
    if (!name || !*name)
        strcpy(buffer, "One for you, one for me.");
    else
        sprintf(buffer, "One for %s, one for me.", name);
}

File: solutions/go/two-fer/two_fer.go
package twofer
import "fmt"

// ShareWith generates a sharing message based on the given name.
// If the input name is an empty string, it defaults to "you".
func ShareWith(name string) string {
    // If the name is an empty string, set it to the default value "you"
    if name == "" {
        name = "you"
    }

    // Use fmt.Sprintf to create a formatted string with the given name
    // and return the result
	return fmt.Sprintf("One for %s, one for me.", name)
}
